addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -70,17 +70,38 @@ weSnippetEditor.Class.include({
                 size: 'medium',
                 title: _t("Google Map API Key"),
                 buttons: [
-                    {text: _t("Save"), classes: 'btn-primary', close: true, click: async () => {
-                        const newAPIKey = dialog.$('#api_key_input').val() || false;
-                        await this._rpc({
-                            model: 'website',
-                            method: 'write',
-                            args: [
-                                [websiteId],
-                                {google_maps_api_key: newAPIKey},
-                            ],
-                        });
-                        invalidated = true;
+                    {text: _t("Save"), classes: 'btn-primary', close: false, click: async (ev) => {
+                        const $apiKeyInput = dialog.$('#api_key_input');
+                        const valueAPIKey = $apiKeyInput.val() || false;
+                        if (!valueAPIKey) {
+                            $apiKeyInput.addClass('is-invalid');
+                            dialog.$('#api_key_help').text(_t('Enter an API Key'));
```

Comment:
Yes and No, the line is duplicated(3 times) but the code can only be accessed once.

Reply:
Not sure what `Yes and No` mean :roll_eyes: 
But even if accessed one in a same flow, use a variable, it will also be easier if it changes later and is a good practice